src/pages/finish/finish.ts

```diff
@@ -32,9 +113,23 @@ export class FinishModalPage {
         this.viewCtrl.dismiss();
       }, 4000);
     }
+    if (this.navParams.data.coin === 'doge' && this.cssClass === 'success') {
```

Comment:
I'd add a control to check if `coin` is undefined. What do you think?

Reply:
yes, that make sense...